### ISSUE TITLE
Custom extension scripts executing again

### ISSUE THREAD

**Issue by deepakvelani** [elastic/azure-marketplace#123]
Hi,

We deployed the cluster using the marketplace template (using elastic version 2.4.1) and customised the kibana.yml and elasticsearch.yml to use different ports, storage account keys for backups etc. It has been working fine with those changes. Last week in one cluster we tried to disable the vm diagnostic logs from the azure portal and by doing that the kibana.yml and elastichseach.ymls got reset to the default (with the settings when a fresh cluster is deployed). Last week in another cluster we enabled the azure VM backup agent for a one off VM backup and noticed the same thing happened here as well. Looking into it further we noticed that the custom scripts are being executed again as the custom scripts extension agent was changed from 2.0.2 to 2.0.3.

Are we ok to remove the custom scripts extension or are those scripts (data-node-install.sh, vm-disk-utils-0.1.sh, kibana-install.sh, elasticsearch-ubuntu-install.sh and user-information.sh) required to there on a running cluster?

thanks
Deepak

**Comment by russcam** [comment]
Custom script extensions can run again when a VM is re-provisioned e.g. if you stop the VM via Azure portal, and start the VM again, the VM will be re-provisioned, so the script will run again. This is part of Azure.

The elasticsearch script has a [measure in place](https://github.com/elastic/azure-marketplace/blob/master/src/scripts/elasticsearch-ubuntu-install.sh#L832-L844) to allow it to be run more than once, but the default elasticsearch.yml configuration that it sets up will be different to the one that you have modified it to.

I wouldn't recommend removing the scripts as I expect this would cause an error within Azure if the VM needs to be re-provisioned. I think it may make the most sense to fork this repository, make your changes, then use your fork to [deploy a cluster from the command line](https://www.elastic.co/blog/elasticsearch-and-kibana-deployments-on-azure).

After forking, if you clone locally and run

```sh
npm install
```

to install all the node packages, then 

```sh
npm run links
```

All of the resource URIs within the repository will be changed to point to your github clone. Then, when deploying from the command line, use

```
https://raw.githubusercontent.com/<USER>/<REPOSITORY>/<GITHUB COMMIT/TAG/BRANCH>/src/mainTemplate.json
```

for the template url and

```
https://raw.githubusercontent.com/<USER>/<REPOSITORY>/<GITHUB COMMIT/TAG/BRANCH>/src
```
for the `artifactsBaseUrl` template parameter value. You'll need to substitute `<USER>`, `<REPOSITORY>` and `<GITHUB COMMIT/TAG/BRANCH>` for the values that make sense for your fork

**Comment by deepakvelani** [comment]
Thanks for the info russcam.

I understand that if i remove the scripts only then there might be an error in azure when it tries to look for the script that are no longer there. What if we removed the custom script extension itself which would then mean Azure wouldnt check for the scripts and would not try to run them.

**Comment by deepakvelani** [comment]
After some digging around we noticed that when we deployed the cluster from the template the custom script agent 2.0.2 got installed. After around a month and half when we did the diagnostic log or backup change they installed an agent on the vm and in the process the custom script agent got updated to 2.0.3. 

When the custom scripts are run they store the seqnum in the path /var/lib/waagent/custom-script/seqnum. The agent checks this file to see if it has to run the custom scripts already. During the upgrade from 2.0.2 to 2.0.3 the process removed the folder /var/lib/waagent/custom-script and recreated it therefore the seqnum file was empty and as a result of that the scripts ran again.

We have removed the custom script extension from the VMs so the scripts are not run again.
